system/util/Util.cfc

```diff
@@ -111,4 +111,36 @@ component{
 		return slug;
 	}
 
-}	
\ No newline at end of file
+
+	/**
+	* Find all methods on a given metadata and it's parents with a given annotation
+	* @annotation The annotation name to look for on methods
+	* @metadata The metadata to search (recursively) for the provided annotation
+	*/
+    public array function getAnnotatedMethods(
+		required string annotation,
+		required struct metadata
+	) {
+        var lifecycleMethods = [];
+
+        if( StructKeyExists( arguments.metadata, "functions" ) ){
+            var funcs = arguments.metadata.functions;
+            lifecycleMethods.addAll( ArrayFilter( funcs, function( func ){
```

Comment:
For testing, I'd think to keep it as compatible as possible. Array Filter should be fairly trivial to reimplement.